altair/vegalite/v5/schema/channels.py

```diff
@@ -38,9 +38,14 @@ def to_dict(self, validate=True, ignore=(), context=None):
                 parsed.pop('type', None)
             elif not (type_in_shorthand or type_defined_explicitly):
                 if isinstance(context.get('data', None), pd.DataFrame):
-                    raise ValueError("{} encoding field is specified without a type; "
-                                     "the type cannot be inferred because it does not "
-                                     "match any column in the data.".format(shorthand))
+                    raise ValueError(
+                        '"{}" cannot be found.'.format(shorthand)
+                        + "\n- If you are trying to reference a column in the dataframe,"
+                        + " there is likely a typo in the column name."
```

Comment:
What about when a URL is being used, like `alt.Chart(data.cars.url)`?  I think that is an important example where the old error message is helpful, because the type is necessary.  Do you see what I mean?

Reply:
That will activate the `else` section below, which still raises the old error message about the missing data type. This new modified error message only shows up if the data is a pandas data frame.

I could not find a way to raise an informative error when there is a typo in the column name in a non-pandas data source and opened https://github.com/altair-viz/altair/issues/2825 to track that.

Reply:
The old message does not contain any reference to "columns", so if you use a URL without a type specified you will see:

> ValueError: Horsepowers encoding field is specified without a type; the type cannot be inferred because it does not match any column in the data.

Is that what you wanted or do you think we should modify the message in this case too?

Reply:
Thanks @joelostblom I misunderstood again.  Everything here looks good to me!